Source/Services/Beam/beam_manager_impl.cpp

```diff
@@ -247,6 +275,88 @@ beam_manager_impl::init_worker(
     return;
 }
 
+
+void xbox::services::beam::beam_manager_impl::process_messages_worker()
+{
+    static const int chunkSize = 10;
+    while (m_processing)
+    {
+        {
+            std::lock_guard<std::recursive_mutex> lock(m_messagesLock);
+            for (int i = 0; i < chunkSize && !m_unhandledFromService.empty(); i++)
+            {
+                std::shared_ptr<beam_rpc_message> currentMessage = m_unhandledFromService.front();
+                m_unhandledFromService.pop();
+                try
+                {
+                    if (currentMessage->m_json.has_field(RPC_TYPE))
+                    {
+                        string_t messageType = currentMessage->m_json[RPC_TYPE].as_string();
+
+                        if (0 == messageType.compare(RPC_REPLY))
+                        {
+                            process_reply(currentMessage->m_json);
+                        }
+                        else if (0 == messageType.compare(RPC_METHOD))
+                        {
+                            process_method(currentMessage->m_json);
+                        }
+                        else
+                        {
+                            LOGS_INFO << "Unexpected message from service";
```

Comment:
ah, good catch. I'll do a pass through on another change to update throughout.

---
In reply to: [117137492](https://github.com/WatchBeam/interactive-cpp/pull/6#discussion_r117137492) [](ancestors = 117137492)